sdk/examples/xo_go/src/sawtooth_xo/handler/handler.go

```diff
@@ -0,0 +1,322 @@
+		{1, 2, 3}, {4, 5, 6}, {7, 8, 9},
+		{1, 4, 7}, {2, 5, 8}, {3, 6, 9},
+		{1, 5, 9}, {3, 5, 7},
+	}
+
+	for _, win := range wins {
+		if board[win[0]-1] == letter && board[win[1]-1] == letter && board[win[2]-1] == letter {
+			return true
+		}
+	}
+	return false
+}
+
+func unpackPayload(payloadData []byte) (*XoPayload, error) {
+	if payloadData == nil {
+		return nil, &processor.InvalidTransactionError{"Must contain payload"}
+	}
+
+	parts := strings.Split(string(payloadData), ",")
+	if len(parts) != 3 {
+		return nil, &processor.InvalidTransactionError{"Payload is malformed"}
+	}
+
+	payload := XoPayload{}
+	payload.Name = parts[0]
+	payload.Action = parts[1]
+
+	if len(payload.Name) < 1 {
+		return nil, &processor.InvalidTransactionError{"Name is required"}
+	}
+
+	if len(payload.Action) < 1 {
+		return nil, &processor.InvalidTransactionError{"Action is required"}
+	}
+
+	if payload.Action == "take" {
+		space, err := strconv.Atoi(parts[2])
+		if err != nil {
+			return nil, &processor.InvalidTransactionError{
+				fmt.Sprintf("Invalid Space: '%v'", parts[2])}
+		}
+		payload.Space = space
+	}
+
+	if strings.Index(payload.Name, "|") != -1 {
+		return nil, &processor.InvalidTransactionError{
+			fmt.Sprintf("Invalid Name (char '|' not allowed): '%v'", parts[2])}
+	}
+
+	return &payload, nil
+}
+
+func unpackHeader(headerData []byte) (*transaction_pb2.TransactionHeader, error) {
+	header := &transaction_pb2.TransactionHeader{}
+	err := proto.Unmarshal(headerData, header)
+	if err != nil {
+		return nil, &processor.InternalError{
+			fmt.Sprint("Failed to unmarshal TransactionHeader: %v", err)}
+	}
+	return header, nil
+}
+
+func unpackGame(gameData []byte) (*Game, error) {
+	parts := strings.Split(string(gameData), ",")
+	if len(parts) != 5 {
+		return nil, &processor.InternalError{
+			fmt.Sprintf("Malformed game data: '%v'", string(gameData))}
+	}
+
+	game := &Game{
+		Name:    parts[0],
+		Board:   parts[1],
+		State:   parts[2],
+		Player1: parts[3],
+		Player2: parts[4],
+	}
+
+	switch game.State {
+	case "P1-WIN", "P2-WIN", "TIE", "P1-NEXT", "P2-NEXT":
+	default:
+		return nil, &processor.InternalError{
+			fmt.Sprintf("Game '%v' has reached invalid state: '%v'", string(gameData), game.State)}
+	}
+
+	return game, nil
+}
+
+func packGame(game *Game) []byte {
+	var buffer bytes.Buffer
+	buffer.WriteString(game.Name)
+	buffer.WriteString(",")
+	buffer.WriteString(game.Board)
+	buffer.WriteString(",")
+	buffer.WriteString(game.State)
+	buffer.WriteString(",")
+	buffer.WriteString(game.Player1)
+	buffer.WriteString(",")
+	buffer.WriteString(game.Player2)
+	return buffer.Bytes()
+}
+
+func loadGame(name string, state *processor.State) (*Game, error) {
+	// Use the namespace prefix + the hash of the game name to create the
+	// storage address
+	address := namespace + hexdigest(name)[:64]
+
+	results, err := state.Get([]string{address})
+	if err != nil {
+		return nil, &processor.InternalError{fmt.Sprint("Error getting state:", err)}
+	}
+
```

Comment:
Missing pipe handling: https://github.com/hyperledger/sawtooth-core/blob/master/sdk/examples/xo_python/sawtooth_xo/processor/handler.py#L105

Also, the state.Get() omits addresses that have data of length 0: https://github.com/hyperledger/sawtooth-core/blob/master/sdk/go/src/sawtooth_sdk/processor/state.go#L113

Should look something like:
```
gameData, exists := results[address]
if !exists {
    return nil, nil
}

games := strings.Split(gameData, "|")
for g := range games {
    game, err := unpackGame(g)
    if err != nil {
        return nil, err
    }
    if game.Name == name {
        return game, nil
    }
}
return nil, nil
```